### PR TITLE
feat: polish package structure

### PR DESCRIPTION
**Pull request by aman00323**
- Remove benchmark (we will cover in test case in next PR)
- Add GitHub action to build and test
- Ignore build files that are inside the example directory.

### CODE REVIEW

**Comment by aman00323** [comment]
@satyrius What's your thought on this?

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.17
```
Comment (satyrius) [inline .github/workflows/go.yml:19]: We should test latest golang version as well. Let's make a matrix with 1.17, 1.18 and 1.19.

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module github.com/satyrius/gonx
+
+go 1.17
+
+require github.com/smartystreets/goconvey v1.7.2
```
Comment (satyrius) [inline go.mod:5]: do we have to add `goconvey` here? it's a dev dependency for tests.

**Review by satyrius — CHANGES_REQUESTED** [review]
Please do not remove benchmarks completely

File: .github/workflows/go.yml (hunk 1 above)
Comment (aman00323) [reply .github/workflows/go.yml]: i have added for other OS support as well, do let me if you have different thought on this.

File: go.mod (hunk 2 above)
Comment (aman00323) [reply go.mod:5]: yes @satyrius it would be included in go.mod

**Comment by aman00323** [comment]
@satyrius I will be adding a benchmark test in another PR. I have removed it as of now for the GitHub action.

please check test for all version and do let me know

**Review by satyrius — APPROVED** [review]

**Comment by satyrius** [comment]
@aman00323 please add benchmarks back in the next PR

**Pull request merged**